### PR TITLE
Fix: don't use deprecated API

### PR DESCRIPTION
**Pull request by mysticatea**
**What is the purpose of this pull request? (put an "X" next to item)**

[X] Other, please explain

**What changes did you make? (Give an overview)**

This PR fixes `relative-module-resolver.js` to not use deprecated API.

In [Node.js v12.2.0](https://nodejs.org/en/blog/release/v12.2.0/), `Module.createRequireFromPath()` function has been deprecated and replaced by `Module.createRequire()` function because it newly accepts URL as well.

**Is there anything you'd like reviewers to focus on?**

Nothing in particular.

### CODE REVIEW

**Comment by not-an-aardvark** [comment]
Does our polyfill actually support the functionality of `createRequire`? It seems like we only support paths but not URLs.

**Comment by mysticatea** [comment]
I think that our use case handles only file paths. The second parameter `relativeToPath` is an absolute path which was processed by `path` module.

The purpose of this PR is just avoiding of use of deprecated API.

File: lib/util/relative-module-resolver.js [hunk 1]
```diff
@@ -8,15 +8,25 @@
 const Module = require("module");
 const path = require("path");
 
-// Polyfill Node's `Module.createRequireFromPath` if not present (added in Node v10.12.0)
-const createRequireFromPath = Module.createRequireFromPath || (filename => {
-    const mod = new Module(filename, null);
+// Polyfill Node's `Module.createRequire` if not present
```
Comment (not-an-aardvark) [inline lib/util/relative-module-resolver.js]:
```suggestion
// Polyfill Node's `Module.createRequire` if not present. We only support the case where the argument is a filepath, not a URL.
```

**Comment by platinumazure** [comment]
Is there a way we could use `no-restricted-syntax` or a custom rule to try to catch the URL case? (I assume this would be difficult but I have to ask.)

**Comment by not-an-aardvark** [comment]
This would obviously be a much bigger change, but I feel like maybe we're getting to the point where we should just start using TypeScript in ESLint core rather than relying on JSDoc annotations and case-specific linting rules.

**Comment by mysticatea** [comment]
That, starting using TypeScript, is a nice idea to me :)
But, for the dogfooding of the basic set, we might be able to consider `//@ts-check` feature with JavaScript.

**Comment by g-plane** [comment]
Will migrate the codebase of ESLint to TypeScript? Or did I misunderstand something?

**Comment by not-an-aardvark** [comment]
I was just bringing it up as an idea, but I don't think we're migrating anything as a result of this issue. (It would probably be better to have a more complete discussion before deciding to migrate anything.)

**Pull request merged**